### PR TITLE
fix: [ANDROAPP-3588] Show/hide Fab button when bottom is reached

### PR DESCRIPTION
**Pull request by JaimeToca**
## Description
Please include a summary of the change and include the related jira issue if it exists.

[ jira issue ](https://jira.dhis2.org/browse/

## Solution description
If this PR is a fix include a brief description on how the issue is solved.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [ ] Smartphone Emulator
- [ ] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [ ] 9.X - 10.X
- [ ] Other
## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: app/src/main/java/org/dhis2/data/forms/dataentry/FormView.kt [hunk 1]
```diff
@@ -76,4 +96,14 @@ class FormView @JvmOverloads constructor(
...
     }
+
+    private fun checkLastItem(): Boolean {
+        val layoutManager =
+            recyclerView.layoutManager as LinearLayoutManager
+        val lastVisiblePosition = layoutManager.findLastVisibleItemPosition()
+        return lastVisiblePosition != -1 && (
+            lastVisiblePosition == adapter.itemCount - 1 ||
+                adapter.getItemViewType(lastVisiblePosition) == R.layout.form_section
+            )
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/data/forms/dataentry/FormView.kt:107]: reviewing how to improve this

Comment (JaimeToca) [reply app/src/main/java/org/dhis2/data/forms/dataentry/FormView.kt:107]: I managed to get a similar behaviour using this: 

private fun checkLastItem(v: View): Boolean {
       
        if (!v.canScrollVertically(DOWN)){
            return true
        }
        return false
}

could I miss anything related to adapter.getItemViewType(lastVisiblePosition) == R.layout.form_section ? Not sure why was needed to be honest. I thought this adapter.itemCount - 1  could be enough

**Review by Balcan — APPROVED** [review]

File: app/src/main/java/org/dhis2/data/forms/dataentry/FormView.kt (hunk 1 above)
Comment (Balcan) [reply app/src/main/java/org/dhis2/data/forms/dataentry/FormView.kt:107]: No, i think this would be enough 👍

Comment (andresmr) [reply app/src/main/java/org/dhis2/data/forms/dataentry/FormView.kt:107]: yes, maybe with `adapter.itemCount - 1` should work. Have you tested it?

**Review by andresmr — APPROVED** [review]

**Pull request merged**